library/core/src/ffi/c_str.rs

```diff
@@ -735,6 +735,11 @@ impl AsRef<CStr> for CStr {
     }
 }
 
+extern "C" {
+    /// Provided by libc or compiler_builtins.
+    fn strlen(s: *const c_char) -> usize;
+}
+
```

Comment:
Can't you just use `const_strlen` from `count` rather than moving this definition out of `const_strlen`? Just to keep one common interface and leave the door open for const eval.

Reply:
Yeah, please change this back and just call `const_strlen`. If it flags errors about `const_eval_select`, https://github.com/rust-lang/rust/pull/127433 will mark that function const stable for internal use and should fix that.

Reply:
I am on board with renaming `const_strlen` -> just `strlen` as Trevor raised in https://github.com/rust-lang/rust/pull/127433#discussion_r1674807047. It is the standard library's general-purpose implementation of strlen, not something that is specific to usage in `const`.

Can be in this PR, or separate one.

Reply:
Tiny change so I just did it https://github.com/rust-lang/rust/pull/127660